Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docrepository/adapter/AdapterComponentDocRepositoryHelper.java

```diff
@@ -192,51 +199,33 @@ String extractClassificationMetadata(
      * @param documentSlots A list of XDS metadata slots
      * @param slotName The name of the slot containing the desired metadata item
      * @param valueIndex For slot multivalued possibilities, the index value desired. If the value is < 0 then all
-     *            values in the value list are returned in a '~' delimited list.
-     * @return Returns the value of the first metadata value with the given metadata name. Null if not present.
+     * values in the value list are returned in a '~' delimited list. @return Returns the value of the first metadata
+     * value with the given metad ata name. Null if not present.
      */
     String extractMetadataFromSlots(List<oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1> documentSlots,
-            String slotName, int valueIndex) {
-        LOG.debug("extractMetadataFromSlots slotname: " + slotName + "; index: " + valueIndex);
+        String slotName, int valueIndex) {
+
+        LOG.debug("extractMetadataFromSlots slotname: {}; index: {}", slotName, valueIndex);
         String slotValue = null;
-        StringBuffer slotValues = null;
-        boolean returnAllValues = false;
-        if (valueIndex < 0) {
-            returnAllValues = true;
-            slotValues = new StringBuffer();
-        }
-        for (oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1 slot : documentSlots) {
+
+        for (SlotType1 slot : documentSlots) {
             if (slotName.equals(slot.getName())) {
-                LOG.debug("Found " + slotName + ": " + slot.getValueList().getValue());
-                if (returnAllValues) {
-                    int listSize = slot.getValueList().getValue().size();
-                    int counter = 0;
-                    Iterator<String> iter = slot.getValueList().getValue().iterator();
-                    while (iter.hasNext()) {
-                        String value = iter.next();
-                        slotValues.append(value);
-                        counter++;
-                        if (counter < listSize) {
-                            slotValues.append(VALUE_LIST_SEPERATOR);
-                        }
-                    }
+                LOG.debug("Found {}: {}", slotName, slot.getValueList().getValue());
```

Comment:
if slot is a not null and slot.getValueList() is a required field, then we just need to remove getValue() part.